### PR TITLE
Ignored flags

### PR DESCRIPTION
**Pull request by Arkaeriit**
GNU rm compatibility as discussed in issue #9.

### CODE REVIEW

**Comment by Arkaeriit** [comment]
I think it is ready to be merged.

File: rem.go [hunk 1]
```diff
@@ -405,17 +424,23 @@ func promptBool(promptStr string) (yes bool) {
 }
 
 func argsHaveOption(long string, short string) (hasOption bool, foundAt int) {
-	for i, arg := range os.Args {
-		if arg == "--"+long || arg == "-"+short {
-			return true, i
-		}
+	hasOption, foundAt = argsHaveOptionExplicit("-" + short)
```
Comment (quackduck) [inline rem.go:427]: bug: this means `argsHaveOption` will always prefer the short option instead of the one that comes last

**Comment by Arkaeriit** [comment]
> By the way, does this PR make the changes #8 does? Or are both separate?

They are two separate PRs, they do not interfere with one another.

File: rem.go [hunk 2]
```diff
@@ -404,18 +423,33 @@ func promptBool(promptStr string) (yes bool) {
...
+	return argsHaveOptionExplicit("--" + long)
+}
+
+func argsHaveOptionExplicit(argFull string) (hasOption bool, foundAt int) {
 	for i, arg := range os.Args {
-		if arg == "--"+long {
+		if arg == "--" {
+			return false, 0
+		}
+		if arg == argFull {
```
Comment (quackduck) [inline rem.go:452]: IMO this is overly complicated. It's okay to have two separate functions for long args and then one for short and long args that both loop through `os.Args`

**Comment by quackduck** [comment]
Fixes #9

**Comment by Arkaeriit** [comment]
I added back arguments after `--` being properly ignored and I merged in this the `-f` flag.

File: rem.go [hunk 3]
```diff
@@ -29,14 +29,21 @@ Options:
...
    --disable-copy         if files are on a different fs, don't rename by copy
+   -f/--force             Do not print error message on inexistant file,
+                          used for compatibility with rm
    -h/--help              print this help message
-   -v/--version           print Rem version`
+   --version              print Rem version
+   -i/--interactive       ignored, used for compatibility with rm
+   -r/-R/--recursive      ignored, used for compatibility with rm
+   -v/--verbose           ignored, used for compatibility with rm
+   --                     all arguments after this as considered files`
```
Comment (quackduck) [inline rem.go:39]: Let's make a note after all the options that says "Rem ignores flags used by `rm` for compatibility. Use `--` when files resemble flags."

We don't need to mention each flag that's ignored.

**Comment by Arkaeriit** [comment]
I pushed the current state of things. The two mode are here.
I still need to rewrite the help messages and I will try to implement the interactive flag in the rm-mode.

**Comment by Arkaeriit** [comment]
I think it is ready to be merged but I am open for any inputs.

**Comment by Arkaeriit** [comment]
It looks like you made some bugs. I will look into them.

**Comment by Arkaeriit** [comment]
It was an easy fix. It looks like it works now.

**Pull request merged**